performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift

```diff
@@ -102,21 +101,51 @@ class Process: ObservableObject {
       await updateStatusAsync(to: .success)
     }
 
-    @available(iOS 15, tvOS 15, *) func uploadImageAsync() async {}
+    @available(iOS 15, tvOS 15, *) func uploadImageAsync(quality: CGFloat = 0.5) async {
+      guard let jpg = image?.jpegData(compressionQuality: quality) else {
+        print("Could not convert image into correct format.")
+        await updateStatusAsync(to: .failure)
+        return
+      }
+
+      let metadata = StorageMetadata()
+      metadata.contentType = "image/jpeg"
+      let reference = Storage.storage().reference().child("image.jpg")
+      await updateStatusAsync(to: .running)
+
+      reference.putData(jpg, metadata: metadata) { metadata, error in
```

Comment:
Use Result type API from FirebaseStorageSwift. See examples in https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseStorageSwift/Tests/Integration/StorageIntegration.swift

Reply:
Done ✅